src/components/Auth/AuthButton/AuthButton.js

```diff
@@ -0,0 +1,53 @@
+import { Button } from '@mui/material';
+import { useAuth0 } from '@auth0/auth0-react';
+import { useDispatch } from 'react-redux';
+import React from 'react';
+import { userLogout } from '../../../reduxStore/userSlice';
+
+const AuthButton = ({
+  type, variant = 'text', onClickCallback = () => {}, isAnchor = false,
+}) => {
+  const { dispatch } = useDispatch();
+  const { loginWithRedirect, logout } = useAuth0();
+
+  const handleLogin = async () => {
+    await loginWithRedirect();
+  };
+
+  const handleSignUp = async () => {
+    await loginWithRedirect({
+      authorizationParams: {
+        screen_hint: 'signup',
+      },
+    });
+  };
+
+  const handleLogout = () => {
+    logout({
+      logoutParams: {
+        returnTo: window.location.origin,
+      },
+    }).then(dispatch(userLogout()));
+  };
+
+  const handleClick = () => {
+    switch (type) {
+      case 'Login':
+        return handleLogin();
+      case 'Signup':
+        return handleSignUp();
+      case 'Logout':
+        return handleLogout();
+      default:
+        return onClickCallback();
+    }
+  };
+
+  return (
+    <Button variant={variant} onClick={handleClick} component={isAnchor ? 'a' : null}>
```

Comment:
we need to remove the "isAnchor" prop and never set the component to an anchor tag.  we can replace that with a "from" prop that can be used in some inline styling to remove borders and set the font to black.  i also think we need some condiditonal styling to change the color of the logout to a red button and have the button be green when its login or sign up.  @mikahpinegar thoughts?

Reply:
yeah i agree with all that. the nav links are needed because they route to a different page on click. how about adding a prop to the button component called isNavButton which will decide whether to return NavLink or Button? then use that component for everything